src/components/shoppingList/index.tsx

```diff
@@ -19,20 +25,28 @@ const ShoppingList = ({purchaseList}: PurchaseListProps): JSX.Element => {
                                 <input
                                     className="purchase-place-input"
                                     defaultValue={purchase.place}
+                                    onChange={(event) =>  purchase.place = event.target.value}
```

Comment:
I just paid attention that you are trying to change the state directly, this is wrong